### DISCUSSION TITLE
Erlang process count keeps growing

### DISCUSSION THREAD

**Discussion by running-g** [rabbitmq/rabbitmq-server#2425] [Other] [answered]
Server:================================================
RabbitMQ server:3.8.6
Erlang 23.0.2  And 22.3
OS Version:Docker And  Ubuntu 7.4.0-1ubuntu1~18.04.1
Cluster Nodes:3
RabbitMQ plugins:rabbitmq_management,rabbitmq_peer_discovery_k8s,rabbitmq_auth_backend_http,rabbitmq_event_exchange,rabbitmq_mqtt,rabbitmq_web_mqtt,rabbitmq_prometheus

Clients:================================================
5000 MQTT Client
20 AMQP Client

If there is no new connection,It will stop growing.
I also tested it at 3.8.5, no problem.

prometheus screenshot：
![5](https://user-images.githubusercontent.com/17699872/89877679-0378b680-dbf3-11ea-9f25-db1351448056.png)

![1](https://user-images.githubusercontent.com/17699872/89877280-733a7180-dbf2-11ea-8126-bf497d580ebb.png)
![2](https://user-images.githubusercontent.com/17699872/89877287-76cdf880-dbf2-11ea-90fa-a80085e92fb4.png)

**Comment by michaelklishin** [comment, marked as answer]
@running-g it is expected that the more connections you have, the more Erlang processes there will be on a node. We don't have a lot of information to work with here. Your clients may be [leaking connections](https://www.rabbitmq.com/connections.html#monitoring) or creating a [high connection churn scenario](https://www.rabbitmq.com/connections.html#high-connection-churn).

What has changed in `3.8.6` is https://github.com/rabbitmq/rabbitmq-mqtt/pull/235, which means MQTT client connections that could fail to be accepted with `3.8.5`, would be accepted with `3.8.6`.

See server logs for clues and what client IDs are used by said connections. A [traffic capture](https://www.rabbitmq.com/amqp-wireshark.html) would provide a lot of highly relevant information as well.

**Comment by michaelklishin** [comment]
We have discovered that MQTT plugin can leave processes behind between `3.8.5` and `3.8.7`, see https://github.com/rabbitmq/rabbitmq-mqtt/pull/239.

**Comment by gerhard** [comment]
Does https://github.com/rabbitmq/rabbitmq-mqtt/pull/239 sound like the issue you were having @running-g?
